be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -707,88 +730,273 @@ void JsonReader::_construct_column(simdjson::ondemand::value& value, Column* col
         }
 
         if (ok) {
-            column->append_strings(literal_1_slice_vector);
+            column->append_datum(literal_1_slice);
         } else {
-            column->append_strings(literal_0_slice_vector);
+            column->append_datum(literal_0_slice);
         }
         break;
     }
 
     case simdjson::ondemand::json_type::number: {
-        simdjson::ondemand::number_type tp;
+        _construct_column_with_number(value, column, type_desc);
+        break;
+    }
+
+    case simdjson::ondemand::json_type::string: {
+        _construct_column_with_string(value, column, type_desc);
+        break;
+    }
+
+    case simdjson::ondemand::json_type::object:
+    case simdjson::ondemand::json_type::array: {
+        std::unique_ptr<char[]> buf;
+        size_t buflen{};
+        JsonFunctions::minify_json_to_string(value, buf, buflen);
+
+        column->append_datum(Slice{buf.get(), buflen});
+        break;
+    }
+
+    default: {
+        column->append_nulls(1);
+        break;
+    }
+    }
+}
+
+void JsonReader::_construct_column_with_number(simdjson::ondemand::value& value, NullableColumn* column,
+                                               const TypeDescriptor& type_desc) {
+    simdjson::ondemand::number_type tp;
+
+    auto err = value.get_number_type().get(tp);
+    if (UNLIKELY(err)) {
+        column->append_nulls(1);
```

Comment:
It's unreasonable to treat errors as NULLs, should filter error rows out, and report the errors rows to the user.